modules/effects/arrow-effects-kotlinx-coroutines-instances/src/main/kotlin/arrow/effects/DeferredKInstances.kt

```diff
@@ -87,7 +87,7 @@ interface DeferredKMonadDeferInstance : MonadDefer<ForDeferredK>, DeferredKBrack
 @extension
 interface DeferredKAsyncInstance : Async<ForDeferredK>, DeferredKMonadDeferInstance {
   override fun <A> async(fa: Proc<A>): DeferredK<A> =
-    DeferredK.async(fa = fa)
+    DeferredK.async(fa = { _, cb -> fa(cb) })
```

Comment:
don't we need to pass in the connection from the callers? looks like we always ask for it in the `async` arguments (`DeferredKProc`) but then it's being instantiated inside the method itself and the param ignored?

Reply:
This is just for now.

This PR just adds the support on the concrete types. This is just in an attempt to do everything in smaller PRs. A discussion of where goes what needs to follow still, I think me & Paco are a little out of sync on this.

Reply:
This seems like the correct approach, yes. The correct implementation (the one in Concurrent) has to have a connection parameter, and the Async one is an specialization where the parameter is ignored.